### PR TITLE
Add WinUI to metadata and sample creation scripts

### PR DESCRIPTION
**Pull request by ZackAllen**
This PR supports WinUI in the following ways:
- Adds support for generating WinUI samples with the `samplegen.py` script.
- Adds support for syncing WinUI metadata and json information with the `process_metadata.py` script.

This sample also:
- Updates sample metadata to use 100.11.2 instead of 100.11.0 across all platforms.

### CODE REVIEW

File: tools/sample_generator/templates/default/WinUI.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    [offline_data_attr]
+    public partial class sample_name
+    {
+        public sample_name()
+        {
+            InitializeComponent();
+            Initialize();
+        }
+
+        private void Initialize()
```
Comment (mbranscomb) [inline tools/sample_generator/templates/default/WinUI.xaml.cs:40]: Should this be async Task?

File: tools/sample_generator/templates/default/WinUI.xaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<UserControl 
+    x:Class="ArcGISRuntime.WinUI.Samples.sample_name.sample_name"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:esri="using:Esri.ArcGISRuntime"
+    xmlns:esriUI="using:Esri.ArcGISRuntime.UI.Controls">
+    <Grid>
+        <esriUI:Geo_View x:Name="MyGeo_View" />
```
Comment (mbranscomb) [inline tools/sample_generator/templates/default/WinUI.xaml:8]: Is `Geo_View` a placeholder?

**Review by mbranscomb — CHANGES_REQUESTED** [review]
@ZackAllen Perhaps just a minor change to the default xaml.cs to change void to async Task.

File: tools/sample_generator/templates/default/WinUI.xaml (hunk 2 above)
Comment (ZackAllen) [reply tools/sample_generator/templates/default/WinUI.xaml:8]: Geo_View gets changed to MapView or SceneView when running the script.

File: tools/sample_generator/templates/default/WinUI.xaml.cs (hunk 1 above)
Comment (ZackAllen) [reply tools/sample_generator/templates/default/WinUI.xaml.cs:40]: There aren't always async calls in initialize, so this should be changed when async calls are used.

Comment (mbranscomb) [reply tools/sample_generator/templates/default/WinUI.xaml.cs:40]: @ZackAllen That's a fair point. Seems like more often than not though, we're creating maps/scenes/layers/symbols and they're all async operations we should be awaiting (or handling load status changed).

**Review by nCastle1 — APPROVED** [review]

**Pull request merged**